contracts/extensions/StakedExpenditure.sol

```diff
@@ -129,17 +132,44 @@ contract StakedExpenditure is ColonyExtensionMeta {
     colony.obligateStake(msgSender(), _domainId, stakeAmount);
     uint256 expenditureId = colony.makeExpenditure(_permissionDomainId, _childSkillIndex, _domainId);
 
-    stakes[expenditureId] = Stake({ creator: msgSender(), amount: stakeAmount });
+    stakes[expenditureId] = Stake({ creator: msgSender(), stagedPayment: _stagedPayment, amount: stakeAmount });
     colony.transferExpenditure(expenditureId, msgSender());
 
-    emit ExpenditureMadeViaStake(msgSender(), expenditureId, stakeAmount);
+    emit ExpenditureMadeViaStake(msgSender(), expenditureId, stakeAmount, _stagedPayment);
+  }
+
+  /// @notice Release a staged payment slot
+  /// @param _permissionDomainId The domainId in which the extension has the arbitration permission
+  /// @param _childSkillIndex The index that the `_expenditureId` is relative to `_permissionDomainId`,
+  /// @param _expenditureId The id of the expenditure
+  /// @param _slot The slot being released
+  function releaseStagedPayment(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    uint256 _expenditureId,
+    uint256 _slot
+  )
+    public
+  {
+    require(stakes[_expenditureId].stagedPayment, "staked-expenditure-not-staged-payment");
+    require(stakes[_expenditureId].creator == msgSender(), "staked-expenditure-not-creator");
+    require(
+      colony.getExpenditure(_expenditureId).status == ColonyDataTypes.ExpenditureStatus.Finalized,
+      "staked-expenditure-not-finalized"
+    );
+
+    bool[] memory mask = new bool[](2); mask[0] = false; mask[1] = true;
+    bytes32[] memory keys = new bytes32[](2); keys[0] = bytes32(0); keys[1] = bytes32(uint256(1));
+    colony.setExpenditureState(_permissionDomainId, _childSkillIndex, _expenditureId, 26, mask, keys, bytes32(0));
+
+    emit StagedPaymentReleased(_expenditureId, _slot);
   }
 
   /// @notice Reclaims the stake if the expenditure is finalized or cancelled
   /// @param _expenditureId The id of the expenditure
   function reclaimStake(uint256 _expenditureId) public {
     Stake storage stake = stakes[_expenditureId];
-    require(stake.creator != address(0x0), "staked-expenditure-nothing-to-claim");
+    require(stake.amount > 0, "staked-expenditure-nothing-to-claim");
```

Comment:
More instructive, certainly - was there another reason for this change?

Reply:
It caught my eye and I realized that the previous approach was suboptimal. I've clearly matured in the months since that was originally written.